server/models/Profile.js

```diff
@@ -0,0 +1,132 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcryptjs");
+
+const profileSchema = new mongoose.Schema({
+  user_username: {
+    type: String,
+    required: true,
+  },
+  first_name: {
+    type: String,
+    required: true,
+  },
+  last_name: {
+    type: String,
+    required: true,
+  },
+  availability: {
+    begin: Date,
+    end: Date,
```

Comment:
As a bonus, it would be nice to see some validations as well (e.g. end > start) for date ranges.